src/Plugins/ComplexCore/src/ComplexCore/Filters/RawBinaryReaderFilter.cpp

```diff
@@ -0,0 +1,203 @@
+std::string RawBinaryReaderFilter::className() const
+{
+  return FilterTraits<RawBinaryReaderFilter>::className;
+}
+
+//------------------------------------------------------------------------------
+Uuid RawBinaryReaderFilter::uuid() const
+{
+  return FilterTraits<RawBinaryReaderFilter>::uuid;
+}
+
+//------------------------------------------------------------------------------
+std::string RawBinaryReaderFilter::humanName() const
+{
+  return "Raw Binary Importer";
+}
+
+//------------------------------------------------------------------------------
+std::vector<std::string> RawBinaryReaderFilter::defaultTags() const
+{
+  return {"#IO", "#Input", "#Read", "#Import"};
+}
+
+//------------------------------------------------------------------------------
+Parameters RawBinaryReaderFilter::parameters() const
+{
+  Parameters params;
+  // Create the parameter descriptors that are needed for this filter
+  params.insert(std::make_unique<FileSystemPathParameter>(k_InputFile_Key, "Input File", "", fs::path(), FileSystemPathParameter::PathType::InputFile));
+  params.insert(std::make_unique<NumericTypeParameter>(k_ScalarType_Key, "Scalar Type", "", NumericType::int8));
+  params.insert(std::make_unique<Int32Parameter>(k_NumberOfComponents_Key, "Number of Components", "", 0));
+  params.insert(std::make_unique<ChoicesParameter>(k_Endian_Key, "Endian", "", 0, ChoicesParameter::Choices{"Little", "Big"}));
+  params.insert(std::make_unique<UInt64Parameter>(k_SkipHeaderBytes_Key, "Skip Header Bytes", "", 0));
+  params.insert(std::make_unique<ArrayCreationParameter>(k_CreatedAttributeArrayPath_Key, "Output Attribute Array", "", DataPath(std::vector<std::string>{"Imported Array"})));
+
+  return params;
+}
+
+//------------------------------------------------------------------------------
+IFilter::UniquePointer RawBinaryReaderFilter::clone() const
+{
+  return std::make_unique<RawBinaryReaderFilter>();
+}
+
+//------------------------------------------------------------------------------
+IFilter::PreflightResult RawBinaryReaderFilter::preflightImpl(const DataStructure& dataStructure, const Arguments& filterArgs, const MessageHandler& messageHandler) const
+{
+  auto pInputFileValue = filterArgs.value<FileSystemPathParameter::ValueType>(k_InputFile_Key);
+  auto pScalarTypeValue = filterArgs.value<NumericType>(k_ScalarType_Key);
+  auto pNumberOfComponentsValue = filterArgs.value<int32>(k_NumberOfComponents_Key);
+  auto pSkipHeaderBytesValue = filterArgs.value<uint64>(k_SkipHeaderBytes_Key);
+  auto pCreatedAttributeArrayPathValue = filterArgs.value<DataPath>(k_CreatedAttributeArrayPath_Key);
+
+  if(pNumberOfComponentsValue < 1)
+  {
+    auto error = MakeErrorResult<OutputActions>(k_RbrZeroComponentsError, "The number of components must be positive.");
+    return {std::move(error), {}};
+  }
+
+  complex::Result<OutputActions> resultOutputActions;
+
+  uint32 inputFileSize = std::filesystem::file_size(pInputFileValue);
```

Comment:
```suggestion
  usize inputFileSize = fs::file_size(pInputFileValue);
```
`fs::file_size` returns a usize not uint32.